### ISSUE TITLE
`visible_messages()` breaks application in certain cases

### ISSUE THREAD

**Issue by mhatvan** [KartikTalwar/gmail.js#667]
I think this occurs when there are ads in a Gmail tab (primary, social, etc):

```
gmailJsLoader.js:3 Uncaught (in promise) TypeError: Cannot read properties of null (reading 'dataset')
    at HTMLTableRowElement.<anonymous> (gmailJsLoader.js:3:50302)
    at Function.each (gmailJsLoader.js:3:60064)
    at b.fn.init.each (gmailJsLoader.js:3:58543)
    at Object.o.dom.visible_messages (gmailJsLoader.js:3:50078)
    at checkForVisibleMessages (extension.js:13:42)
    at addCustomDOMContent (extension.js:155:27)
    at extension.js:100:13
    at c (m=b:869:118)
```

**Comment by josteink** [comment]
Hey there and thanks for reporting!

Looking at the stack trace, is this just using the plain boilerplate project, or something more.. “orchestrated”? 🙂

**Comment by mhatvan** [comment]
That's the plain gmail.js code in a file. I was able to fix this locally by adding checks to the source code for that method.

**Comment by josteink** [comment]
Right. In which case, feel free to submit a PR 😄

**Comment by josteink** [comment]
This should now be published in `v1.0.17` on npmjs.com.
